tests: flash : frdm_mcxn947/mcxn947/cpu0 : tests/drivers/flash/common build failure

**Describe the bug**

below tests build failure on frdm_mcxn947/mcxn947/cpu0


CASE_NAME | ZEPHYR_VERSION | RESULT | PLATFORM | TEST_PATH
-- | -- | -- | -- | --
mgmt.mcumgr.fs.mgmt.hash.supported.crc32 | v3.7.0-3233-g4cc3134aa1c1 | Build failure | frdm_mcxn947/mcxn947/cpu0 | tests/subsys/mgmt/mcumgr/fs_mgmt_hash_supported
mgmt.mcumgr.fs.mgmt.hash.supported.sha256 | v3.7.0-3233-g4cc3134aa1c1 | Build failure | frdm_mcxn947/mcxn947/cpu0 | tests/subsys/mgmt/mcumgr/fs_mgmt_hash_supported
mgmt.mcumgr.fs.mgmt.hash.supported.all | v3.7.0-3233-g4cc3134aa1c1 | Build failure | frdm_mcxn947/mcxn947/cpu0 | tests/subsys/mgmt/mcumgr/fs_mgmt_hash_supported
drivers.flash.common.default | v3.7.0-3233-g4cc3134aa1c1 | Build failure | frdm_mcxn947/mcxn947/cpu0 | tests/drivers/flash/common
sample.usb.dfu | v3.7.0-3233-g4cc3134aa1c1 | Build failure | frdm_mcxn947/mcxn947/cpu0 | samples/subsys/usb/dfu
sample.usb.dfu.permanent.download | v3.7.0-3233-g4cc3134aa1c1 | Build failure | frdm_mcxn947/mcxn947/cpu0 | samples/subsys/usb/dfu
sample.drivers.flash.shell | v3.7.0-3233-g4cc3134aa1c1 | Build failure | frdm_mcxn947/mcxn947/cpu0 | samples/drivers/flash_shell


Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - frdm_mcxn947/mcxn947/cpu0
 - What have you tried to diagnose or workaround this issue?
 - git bisect
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?

```
cbee39ef711e13c2acf1fae090e9c0c123986af2 is the first bad commit
commit cbee39ef711e13c2acf1fae090e9c0c123986af2
Author: Declan Snyder <declan.snyder@nxp.com>
Date:   Thu Aug 8 14:37:15 2024 -0500

    soc: nxp: Add MCXW71
    
    Add MCXW71 SOC, which inherits some qualitiies
    of kinetis heritage platforms.
    
    Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

 soc/nxp/mcx/mcxw/CMakeLists.txt         |   9 ++
 soc/nxp/mcx/mcxw/Kconfig                |  16 +++
 soc/nxp/mcx/mcxw/Kconfig.defconfig      |  12 +++
 soc/nxp/mcx/mcxw/Kconfig.soc            |  22 +++++
 soc/nxp/mcx/mcxw/linker.ld              |  25 +++++
 soc/nxp/mcx/mcxw/mcxw71_platform_init.S |  59 ++++++++++++
 soc/nxp/mcx/mcxw/pinctrl_soc.h          |   7 ++
 soc/nxp/mcx/mcxw/soc.c                  | 166 ++++++++++++++++++++++++++++++++
 soc/nxp/mcx/mcxw/soc.h                  |  13 +++
 soc/nxp/mcx/soc.yml                     |   3 +
 west.yml                                |   2 +-
 11 files changed, 333 insertions(+), 1 deletion(-)
 create mode 100644 soc/nxp/mcx/mcxw/CMakeLists.txt
 create mode 100644 soc/nxp/mcx/mcxw/Kconfig
 create mode 100644 soc/nxp/mcx/mcxw/Kconfig.defconfig
 create mode 100644 soc/nxp/mcx/mcxw/Kconfig.soc
 create mode 100644 soc/nxp/mcx/mcxw/linker.ld
 create mode 100644 soc/nxp/mcx/mcxw/mcxw71_platform_init.S
 create mode 100644 soc/nxp/mcx/mcxw/pinctrl_soc.h
 create mode 100644 soc/nxp/mcx/mcxw/soc.c
 create mode 100644 soc/nxp/mcx/mcxw/soc.h
```

**To Reproduce**

scripts/twister -p frdm_mcxn947/mcxn947/cpu0 -T tests/drivers/flash/common

**Expected behavior**

build pass

**Impact**

flash function for this platform

**Logs and console output**

```
/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/flash/soc_flash_mcux.c
2024-09-21 14:09:40,620 - __main__ -zephyr_ci_build_deploy_test.py : 185 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/flash/soc_flash_mcux.c:53:10: fatal error: fsl_flash.h: No such file or directory
2024-09-21 14:09:40,620 - __main__ -zephyr_ci_build_deploy_test.py : 185 - INFO - 53 | #include "fsl_flash.h"
2024-09-21 14:09:40,620 - __main__ -zephyr_ci_build_deploy_test.py : 185 - INFO - |          ^~~~~~~~~~~~~
2024-09-21 14:09:40,620 - __main__ -zephyr_ci_build_deploy_test.py : 185 - INFO - compilation terminated.
2024-09-21 14:09:40,620 - __main__ -zephyr_ci_build_deploy_test.py : 185 - INFO - ninja: build stopped: subcommand failed.
2024-09-21 14:09:40,620 - __main__ -zephyr_ci_build_deploy_test.py : 185 - INFO - 
2024-09-21 14:09:40,620 - __main__ -zephyr_ci_build_deploy_test.py : 185 - INFO - INFO    - 
```

**Environment (please complete the following information):**

 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used:  v3.7.0-3233-g4cc3134aa1c1

--- a/west.yml
+++ b/west.yml
@@ -198,7 +198,7 @@ manifest:
       groups:
         - hal
     - name: hal_nxp
-      revision: f4e26fad2cfd8b8e8988e835a28667573ed072cf
+      revision: adfa0bbba6f5d36352d387527bdc486616c9f521
       path: modules/hal/nxp
       groups:
         - hal

